test/e2e/operator_test.go

```diff
@@ -158,11 +107,316 @@ func ensureCredentialsRequest() error {
 	return nil
 }
 
+func newAWSLoadBalancerController(name types.NamespacedName, ingressClass string, addons []albo.AWSAddon) albo.AWSLoadBalancerController {
+	return albo.AWSLoadBalancerController{
+		ObjectMeta: v1.ObjectMeta{
+			Name:      name.Name,
+			Namespace: name.Namespace,
+		},
+		Spec: albo.AWSLoadBalancerControllerSpec{
+			SubnetTagging: albo.AutoSubnetTaggingPolicy,
+			IngressClass:  ingressClass,
+			EnabledAddons: addons,
+		},
+	}
+}
+
+func TestMain(m *testing.M) {
+	kubeConfig, err := config.GetConfig()
+	if err != nil {
+		fmt.Printf("failed to get kube config: %s\n", err)
+		os.Exit(1)
+	}
+	cl, err := client.New(kubeConfig, client.Options{})
+	if err != nil {
+		fmt.Printf("failed to create kube client: %s\n", err)
+		os.Exit(1)
+	}
+	kubeClient = cl
+
+	if err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, &infraConfig); err != nil {
+		fmt.Printf("failed to get infrastructure config: %v\n", err)
+		os.Exit(1)
+	}
+
+	kubeClientSet, err = kubernetes.NewForConfig(kubeConfig)
+	if err != nil {
+		fmt.Printf("failed to create kube clientset: %s\n", err)
+		os.Exit(1)
+	}
+
+	if err = ensureCredentialsRequest(); err != nil {
+		fmt.Printf("failed to create credentialsrequest for operator due to: %v\n", err)
+	}
+
+	os.Exit(m.Run())
+}
+
+func getOperator(t *testing.T) {
+	expected := []appsv1.DeploymentCondition{
+		{Type: appsv1.DeploymentAvailable, Status: corev1.ConditionTrue},
+	}
+	name := types.NamespacedName{
+		Name:      operatorName,
+		Namespace: operatorNamespace,
+	}
+	if err := waitForDeploymentStatusConditionUntilN(t, kubeClient, defaultTimeout, defaultCount, name, expected...); err != nil {
+		t.Fatalf("Did not get expected available condition: %v", err)
+	}
+}
+
 func TestOperatorAvailable(t *testing.T) {
+	getOperator(t)
+}
+
+// TestAWSLoadBalancerControllerWithDefaultIngressClass tests the basic happy flow for the operator, mostly
+// using the default values.
+func TestAWSLoadBalancerControllerWithDefaultIngressClass(t *testing.T) {
+	t.Log("Verifying operator availability due to restarts caused by deletion of resources")
+	getOperator(t)
+
+	t.Log("Creating aws load balancer controller instance with default ingress class")
+
+	name := types.NamespacedName{Name: "cluster", Namespace: "aws-load-balancer-operator"}
+	alb := newAWSLoadBalancerController(name, "alb", []albo.AWSAddon{})
+	if err := kubeClient.Create(context.TODO(), &alb); err != nil {
+		if errors.IsAlreadyExists(err) {
```

Comment:
We should delete the controller resource after every test. And it should be recreated from scratch in every test. If it already exists then it should not be updated. Because that's a whole different kind of test, where we ensure that updating the configuration yields a properly configured controller deployment.